tools/yulPhaser/SimulationRNG.cpp

```diff
@@ -35,16 +35,16 @@ bool SimulationRNG::bernoulliTrial(double _successProbability)
 	return static_cast<bool>(distribution(s_generator));
 }
 
-uint32_t SimulationRNG::uniformInt(uint32_t _min, uint32_t _max)
+size_t SimulationRNG::uniformInt(size_t _min, size_t _max)
 {
-	boost::random::uniform_int_distribution<> distribution(_min, _max);
-	return distribution(s_generator);
+	boost::random::uniform_int_distribution<> distribution(static_cast<int>(_min), static_cast<int>(_max));
```

Comment:
It it possible to use a template parameter instead of casting, `uniform_int_distribution<size_t>`?

```
template<typename IntType = int> 
class uniform_int_distribution {
```

Reply:
Distribution (both uniform and binomial) doesn't work with unsigned types as there is an error when resolving `abs(unsigned)`

Reply:
Anyway, this is now part of #9091